feat: add deprecated field to feature flag schema (#410)
launchdarkly/terraform-provider-launchdarkly: docs/resources/feature_flag.md

- `client_side_availability` (Block List) (see [below for nested schema](#nestedblock--client_side_availability))
- `custom_properties` (Block Set, Max: 64) List of nested blocks describing the feature flag's [custom properties](https://docs.launchdarkly.com/home/connecting/custom-properties) (see [below for nested schema](#nestedblock--custom_properties))
- `defaults` (Block List, Max: 1) A block containing the indices of the variations to be used as the default on and off variations in all new environments. Flag configurations in existing environments will not be changed nor updated if the configuration block is removed. (see [below for nested schema](#nestedblock--defaults))

> - `deprecated` (Boolean) Specifies whether the flag is deprecated or not. Note that you cannot create a new flag that is deprecated, but can update a flag to be deprecated.

Reviewer: ```suggestion
- `deprecated` (Boolean) Specifies whether the flag is deprecated or not. You cannot create a new flag that is deprecated, but you can deprecate an existing flag.
```
